### PR TITLE
Validate indexes of queue space in driver.new() method

### PR DESCRIPTION
**Pull request by olmiik**
Currently for all queue implementations  `driver.new()` does not validate queue space, created by `driver.create_space()` method. However corrupted space can lead to unexpected behavior (for example, see #134),

This one adds check to ensure, that indexes, created by `driver.create_space()`, really exist.

### CODE REVIEW

File: queue/abstract/driver/fifo.lua [hunk 1]
```diff
@@ -33,8 +33,21 @@ function tube.create_space(space_name, opts)
     return space
 end
 
+-- validate space of queue
+local function validate_space(space)
```
Comment (LeonidVas) [inline queue/abstract/driver/fifo.lua]: I agree, WET practice can be used here.
But I suggest using something like:
"
local function validate_space(space, indexes)
	local indexes = {'task_id', 'status'}
	for _, index in pairs(indexes) do
		if space.index[index] == nil then
			error('space does not have ' .. index .. ' index')
		end
	end
"

File: t/010-fifo.t [hunk 2]
```diff
@@ -292,6 +292,38 @@ test:test('if_not_exists test', function(test)
...
+    local fifo = require('queue.abstract.driver.fifo')
+    local space = fifo.create_space('corrupted_fifo_space', { engine = engine })
+    
+    local task_id = space.index.task_id
+    local status = space.index.status
+
+    test:test('task_id index does not exist', function(test)
+        test:plan(2)
+
+        space.index.task_id = nil
```
Comment (Totktonada) [inline t/010-fifo.t]: This is not proper use of the index lua api and it is not guaranteed to work in a future. OTOH, I don't see anything that would block usage of \<index_object\>:drop() and then re-creation of the space for a next test.

**Comment by Totktonada** [comment]
@olmiik I want to kindly remind you about this PR. There are a couple of small comments. It would be good to resolve them and finally fix the problem for users.

**Comment by LeonidVas** [comment]
Continued in PR https://github.com/tarantool/queue/pull/136

**Pull request closed without merging**